### PR TITLE
Add event package for working with events

### PR DESCRIPTION
**Pull request by jnicklas**
Working with event emitters has been a really common need when working with Effection. Over time we evolved some helpers for making this more efficient.

We have also extensively used mailboxes for working with events. However, with the advent of resources, it is now pretty easy to implement and efficient and easy to use event iterator. I've called this operation `on`. This is a more light-weight solution than mailboxes. There is some overlap in functionality, but I think this is acceptable. Basically we could replace a lot of the mailboxes in bigtest with this, and it would make it clearer where we actually need mailboxes and where we just use them to subscribe to events.

### CODE REVIEW

**Comment by jnicklas** [comment]
There is an example of the new `on` operation in the [README](https://github.com/thefrontside/effection.js/blob/a2ebf310414e034c76a017d455852ab9b5def0e6/packages/events/README.md).

**Comment by jnicklas** [comment]
It seems that NodeCodeFormatter stripped a ton of dependencies from `yarn.lock` which seems... not good?

File: package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
   "private": true,
   "workspaces": {
     "packages": [
-      "packages/*"
+      "packages/effection/pkg",
```
Comment (jnicklas) [inline package.json:11]: Had to do this to be able to use the main effection package from `events`, not sure if this is correct or if there is a better solution?

Comment (cowboyd) [reply package.json:11]: Yup, until we use yarn2 which _should_ resolve internal packages in dependency order, this is the only way I know. /cc @taras

File: packages/events/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Node.js style EventEmitters (`on`/`off`) and with browser style EventTarget
+(`addEventListeners`/`removeEventListener`).
+
+## Usage
+
+The `once` operation can be used to wait for an event to occur exactly once.
+
+``` typescript
+import { main } from '@effection/node';
+import { once } from '@effection/node';
```
Comment (cowboyd) [inline packages/events/README.md:13]:
```suggestion
import { once } from '@effection/events';
```

File: packages/events/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  yield once(window, "DOMContentLoaded");
+  console.log('Hello World!');
+});
+```
+
+The `on` operation returns a subscription which can be used to loop over events:
+
+``` typescript
+import { main } from '@effection/node';
+import { on } from '@effection/node';
```
Comment (cowboyd) [inline packages/events/README.md:25]:
```suggestion
import { on } from '@effection/events';
```

File: packages/events/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+It is a common pattern, primarily in Node to have an event called `error` which
+has an error as its first argument. `@effection/events` includes a helper to
+make handling this situation convenient:
+
+``` typescript
+import { main } from '@effection/node';
+import { once, throwOnErrorEvent } from '@effection/node';
+
+main(function* sayHello() {
+  let someSocket();
```
Comment (cowboyd) [inline packages/events/README.md:46]:
```suggestion
  let someSocket = createSocket();
```

File: packages/events/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  yield throwOnErrorEvent(someSocket);
+
+  yield once(someSocket, "open");
+
+  let messages = yield once(someSocket, "message");
+
+  while(true) {
+    let message = yield someSocket.next();
```
Comment (cowboyd) [inline packages/events/README.md:55]:
```suggestion
    let message = yield messages.next();
```

File: packages/events/src/on.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      removeListener(this.source, this.eventName, listener);
+    }
+
+  }
+}
+
+
+export function* on<T extends Array<unknown> = unknown[]>(source: EventSource, name: string): Operation<Subscription<T>> {
+  let subscription = new Subscription<T>(source, name);
+  return yield resource(subscription, subscription.subscribe());
```
Comment (cowboyd) [inline packages/events/src/on.ts:34]: very nice 😍

**Comment by cowboyd** [comment]
> It seems that NodeCodeFormatter stripped a ton of dependencies from `yarn.lock` which seems... not good?

It says very clearly in our terms of service that bots may not drink on the job. We may have to fire NodeCodeFormatter.

**Pull request merged**